### PR TITLE
Simplified parameter and block's port setup, improved Simulink signal propagation

### PR DESCRIPTION
**Pull request by diegoferigo**
This is the second PR in preparation of release `4`.

- Improved `parseParameters` method as described in https://github.com/robotology/wb-toolbox/issues/109
- Setting of input / output ports is now easier as described in https://github.com/robotology/wb-toolbox/issues/109
- Minor improvements to `BlockInterface` method's arguments
- Improved Simulink signal propagation (cc @gabrielenava @S-Dafarra compiling seems faster, let me know if you notice improvements also on your machine)

### CODE REVIEW

**Comment by diegoferigo** [comment]
Merging as soon as test pass.

File: toolbox/src/DiscreteFilter.cpp [hunk 1]
```diff
@@ -54,85 +71,59 @@ unsigned DiscreteFilter::numberOfParameters()
...
-    ok = ok && blockInfo->addParameterMetadata(paramMD_init_u0);
-
-    if (!ok) {
-        wbtError << "Failed to store parameters metadata.";
-        return false;
+    const auto& DynPar = ParameterMetadata::DynamicSize;
+
+    const std::vector<ParameterMetadata> metadata{
+        {ParameterType::STRUCT_DOUBLE, ParamIndex::FilterStruct, 1, 1, "Fc"},
+        {ParameterType::STRUCT_DOUBLE, ParamIndex::FilterStruct, 1, 1, "Ts"},
```
Comment (traversaro) [inline toolbox/src/DiscreteFilter.cpp]: `MedianOrder` and `FilterType` have been removed?

File: toolbox/src/GetLimits.cpp [hunk 2]
```diff
@@ -157,137 +174,159 @@ bool GetLimits::initialize(BlockInformation* blockInfo)
...
+                        wbtError << "Failed to get joint limits from the URDF model "
+                                 << "for the joint " << joint + ".";
+                        return false;
+                    }
+                    pImpl->limits.min[i] = min;
+                    pImpl->limits.max[i] = max;
+                }
+            }
+        }
+        // TODO: other limits from the model?
```
Comment (traversaro) [inline toolbox/src/GetLimits.cpp]: Related issue: https://github.com/robotology/idyntree/issues/384 .

File: toolbox/src/DiscreteFilter.cpp (hunk 1 above)
Comment (diegoferigo) [reply toolbox/src/DiscreteFilter.cpp]: You caught this well Silvio, I realized it running some test controllers and I already have modifications on a local branch ;)

**Pull request merged**